### ISSUE TITLE
[Bug]Original message not found error on ios14

### ISSUE THREAD

**Issue by g-savitha** [signalapp/Signal-iOS#4684]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Getting an "original message not found error" when clicked on an image instead of navigating to the image. Please find the attached recording for reference

https://user-images.githubusercontent.com/31612459/104119960-53ea7080-5359-11eb-9623-d6ffa6193c4b.mp4

.

**Actual result:** Doesn't display the image and throws an error 

**Expected result:** Should navigate to the image 

### Device info

**Device**: iphone11

**iOS version**: ios14

**Comment by RomeoGolfBravo** [comment]
This looks as though the Group you are in has ‘Disappearing messages’ activated. The time has elapsed, therefore the original message is no longer viewable - except for in the quoted message. 

I think it should disappear completely, but that’s for the feedback session as it’s designed that way.

**Comment by g-savitha** [comment]
Hey @RomeoGolfBravo, 

No, Disappearing messages is not enabled. This feature is working as expected in android. This is happening only in ios

**Comment by sa1** [comment]
https://debuglogs.org/9994102a98453a009f872d60445689557888555af2486a683a3eb272acde8e4d.zip

I'm seeing same bug. No disappearing messages

**Comment by 211217613** [comment]
https://debuglogs.org/4e10123b0a2f9fc50fdae96805268e111aa862b7f386b6059aaf22d1f00dab7d.zip


same issue.

**Comment by paul-grassl** [comment]
Same issue. I'm experiencing it with quoted links. So the person I'm messaging with quotes a link i sent and it shows to me as "original message not found ". We don't have disappearing messages turned on. 

https://debuglogs.org/aaafd8af2f46f0edd95e73bb4cbcfff5ba703bd21b1cb740cde3602b882ef826.zip

**Comment by sa1** [comment]
Seems to happen when messages are received in background, and not when received when the app is open. (based on limited amount of times I've seen)

**Comment by Pancham97** [comment]
I'm facing the same problem when texting someone in person. It happens to me for media only.

Device: iPhone SE 2020
Platform: iOS 14.4
Signal version: 5.4.0.17

**Comment by ZeroMSN** [comment]
I'm facing the same issue on iPhone 12 Pro Max with iOS 14.4. Whenever I share a picture or a video to a group on signal, I don't see a that file but the recipients do. If someone replies to that message then I see the message 'message not found' within that reply. It's felt winding me up... it's been happening ever since I've had signal installed in my iPhone.

**Comment by maartenhunink** [comment]
Same or similar issue here. 

When an image is quoted, it sometimes shows the 'Original message not found' error in group messages. In the example below the item that is quotes is visible right above, so the original message is there for sure. The same message does show up correctly on desktop.

![IMG_6171](https://user-images.githubusercontent.com/106182/126891784-38d5441c-7362-47f9-a3bd-39dc61558dd6.PNG)

**Comment by ZeroMSN** [comment]
> Same or similar issue here. 
> 
> 
> 
> When an image is quoted, it sometimes shows the 'Original message not found' error in group messages. In the example below the item that is quotes is visible right above, so the original message is there for sure. The same message does show up correctly on desktop.
> 
> 
> 
> ![IMG_6171](https://user-images.githubusercontent.com/106182/126891784-38d5441c-7362-47f9-a3bd-39dc61558dd6.PNG)
> 
> 

That's it. That's the issue. Thanks.

**Comment by guimspace** [comment]
This bug is **reproducible** with consistency.

This bug occurs with these states:
- Signal isn't in the background (removed from the recent, updated or settings changed).
- Signal is closed but in the background and the network changes (Wi-Fi to cellular, vice-versa).

With that, if an user sends a media (image or video) file and someone (including the sender) replies to that media, the reply will have the error message.

This bugs occurs with "Note to self" media messages too if you use the desktop to send media to yourself and reply.

This bug doesn't affect the desktop version.

**Comment by kaktusztea** [comment]
Similar issue:
I have a group where I sent a photo gallery message (without text): 1628443541538.
Later in the same group I made a reply _to my own_ photo-gallery message and wrote a text in it (1628445598095).
After I sent it and left the group view, when I went back, in this last message by me there was an “original message not found” warning, no index picture and link to the photo-gallery message.
On desktop everything was OK, no such issue.

iOS 5.17.1.2 log:
https://debuglogs.org/71e492238274cad15449f520f6fffb09030b396232ca200968031fac6907f55f.zip

My pictures-only photo-gallery message:
1628443541538

My reply message with “original message not found”:
1628445598095

Desktop 5.12.2 (OSX) log:
https://debuglogs.org/a054cb284fbc6514eb93a17ef9f57a83cfa0f3ad5117060fa53c1a3e43d7d527.gz
